astropy/coordinates/representation.py

```diff
@@ -483,7 +554,141 @@ def represent_as(self, other_class):
             return self
         else:
             # The default is to convert via cartesian coordinates
-            return other_class.from_cartesian(self.to_cartesian())
+            if isiterable(other_class):
+                if isinstance(other_class, six.string_types):
+                    raise ValueError('Input to a representation\'s represent_as'
+                                     ' must be a class, not a string. For '
+                                     'strings, use frame objects')
+                if (self.differentials is None or
+                        len(other_class) != 1+len(self.differentials)):
+                    raise ValueError("If multiple classes are passed in to "
+                                     "change the representations of attached "
+                                     "differentials, the number of classes must "
+                                     "be equal to 1 + the number of "
+                                     "differentials. (got {0})"
+                                     .format(len(other_class)))
+
+                new_rep = other_class[0].from_cartesian(self.to_cartesian())
+
+                new_diffs = []
+                for i,diff in enumerate(self.differentials):
+                    new_diffs.append(diff.represent_as(other_class[1+i],
+                                                       base=self))
+                new_rep._differentials = tuple(new_diffs)
+
+            else:
+                new_rep = other_class.from_cartesian(self.to_cartesian())
+
+            return new_rep
+
+    def with_differentials(self, differentials):
+        """
+        Create a new representation with the same positions as this
+        representation, but with new differentials.
+
+        Parameters
+        ----------
+        differentials : Sequence of `~astropy.coordinates.BaseDifferential`
+            The differentials for the new representation to have.
+
+        Returns
+        -------
+        newrepr
+            A copy of this representation, but with the ``differentials`` as
+            its differentials.
+        """
+        # Implemented this way for (1) performance (skips going through the
+        # initializer), and (2) will preserve any additional attributes a user
+        # may have set
+
+        new_differentials = self._validate_differentials(differentials)
+
+        # save the differentials, then strip them so that the copy doesn't do
+        # an unneeded copy, then re-attach them to the right objects
+        olddiffs = self._differentials
+        try:
+            self._differentials = None
+            newrepr = deepcopy(self)
```

Comment:
Why a `deepcopy` - since representations are nominally immutable, I think a shallow copy is OK.

But, really, this to me is premature optimization. I think it is better to just use the initializer, with `copy=False` (which means it is not all that slow), and then speed up everything by defining a no-check initializer option (e.g., `copy=None`?); see #5598 for initial work in that direction.